### ISSUE TITLE
ArgumentError: cannot push_patch/2 because the given path does not point to the current root view (although it's the same root view)

### ISSUE THREAD

**Issue by woylie** [phoenixframework/phoenix_live_view#3556]
### Environment

* Elixir version (elixir -v): Elixir 1.17.3 (compiled with Erlang/OTP 27)
* Phoenix version (mix deps): 1.7.17
* Phoenix LiveView version (mix deps): 1.0.0
* Operating system: Debian
* Browsers you attempted to reproduce this bug on (the more the merrier): 
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: yes

### Actual behavior

Since we updated from LiveView 1.0.0-rc.7 to rc.8/9 and then 1.0.0, we are seeing occasional ArgumentErrors similar to this in production:

```
ArgumentError: cannot push_patch/2 to %URI{scheme: "https", userinfo: nil, host: "somehost", port: 443, path: "/some/path", query: "return_to=%2F", fragment: nil} because the given path does not point to the current root view MyAppWeb.SomeViewLive
```

This happens quite sporadically so far, and with different paths each time. We confirmed that the `push_patch` calls originate from the same LiveView module that is reported, and that the source and destination routes are within the same `live_session`.

We haven't been able to reproduce this in any way. The views work as expected when testing them manually in the browser or in LiveView tests.

I wonder whether this is related to these changes: https://github.com/phoenixframework/phoenix_live_view/pull/3521/files#diff-7f430395ff207cba74ace585af4eee7564f139403ed89378f860582731d5b7c8R36. I don't know how the `vsn` works that is being matched on.

### Expected behavior

No errors are raised.

**Comment by SteffenDE** [comment]
Are you really sure that they are in the same live_session? (We need to adjust the error message).

The vsn is set at compile time so it should never be a problem.

**Comment by josevalim** [comment]
For clarity, is this also same host, same scheme, same authority, same port?

**Comment by SteffenDE** [comment]
Just seen this myself when developing locally, but cannot reproduce, gone after a reload :(

**Comment by wkirschbaum** [comment]
We are seeing the same issue, but only on pages with Flop ( so far ). It happens every couple of times per hour on production for us, but we don't see any visual issues for these cases. The stacktrace from appsignal: 

```
<h1 class="c-box__header">Backtrace</h1>
lib/phoenix_live_view/channel.ex:888 Phoenix.LiveView.Channel.patch_params_and_action!/2 |  
-- | --
lib/phoenix_live_view/channel.ex:856 Phoenix.LiveView.Channel.handle_redirect/4 |  
gen_server.erl:2345 :gen_server.try_handle_info/3 |  
gen_server.erl:2433 :gen_server.handle_msg/6 |  
proc_lib.erl:329 :proc_lib.init_p_do_apply/3

```

We are on the latest deps.

**Comment by josevalim** [comment]
@SteffenDE I have looked at the code and I can see why it would happen in development. In development, we can recompile the code/router while the live view session is still open. That would cause the version to change. My suggestion is to remove the version check, given the connection is open, it should only happen on code reload anyway?

**Comment by SteffenDE** [comment]
> In development, we can recompile the code/router while the live view session is still open.

@josevalim that's what I was thinking too. With hot code updates the version might differ. But it doesn't explain why it would be a problem in production. It also doesn't explain why it doesn't seem to ever happen for push_navigate, where we also compare the versions https://github.com/phoenixframework/phoenix_live_view/blob/1249383ed2b3253850fa899658c9bdf593829c69/lib/phoenix_live_view/session.ex#L24

As some people see this very frequently, we can maybe create a branch that has some logging in case the live_session differs to better see what's going on?

**Comment by josevalim** [comment]
Also, navigate does end-up navigating anyway, there are no crashes, so it is understandable we have not been hearing about it.

**Comment by SteffenDE** [comment]
Closed via https://github.com/phoenixframework/phoenix_live_view/pull/3570.

tl;dr: indeed the issue was clients connecting with a token from an older deploy. As the `live_session_vsn` is included in the signed token, this would cause a mismatch on push_patch. `push_navigate` would fallback to a regular redirect without crashing.

**Comment by chrismccord** [comment]
1.0.1 is out. Thanks!
